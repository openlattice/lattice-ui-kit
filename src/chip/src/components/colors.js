import * as Colors from '../../../colors';

const {
  NEUTRAL,
  PURPLE,
  BLUE,
  TEAL,
  GREEN,
  YELLOW,
  ORANGE,
  RED,
  MAGENTA,
  VIOLET
} = Colors;

export default {
  default: {
    background: NEUTRAL.N00,
    color: NEUTRAL.N600
  },
  blue: {
    background: BLUE.B00,
    color: BLUE.B400,
    clickable: BLUE.B100
  },
  green: {
    background: GREEN.G00,
    color: GREEN.G400,
    clickable: GREEN.G100
  },
  magenta: {
    background: MAGENTA.M00,
    color: MAGENTA.M400,
    clickable: MAGENTA.M100
  },
  orange: {
    background: ORANGE.O00,
    color: ORANGE.O500,
    clickable: ORANGE.O100
  },
  purple: {
    background: PURPLE.P00,
    color: PURPLE.P400,
    clickable: PURPLE.P100
  },
  red: {
    background: RED.R00,
    color: RED.R400,
    clickable: RED.R100
  },
  teal: {
    background: TEAL.T00,
    color: TEAL.T500,
    clickable: TEAL.T100
  },
  violet: {
    background: VIOLET.V00,
    color: VIOLET.V400,
    clickable: VIOLET.V100
  },
  yellow: {
    background: YELLOW.Y00,
    color: YELLOW.Y500,
    clickable: YELLOW.Y100
  },
  primary: {
    color: 'white'
  },
  secondary: {
    color: PURPLE.P400
  },
};
