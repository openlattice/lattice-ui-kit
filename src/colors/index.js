/*
 * @flow
 */

/*
 * Inspiration:
 * https://atlassian.design/guidelines/brand/color-1
 * https://bitbucket.org/atlassian/atlaskit-mk-2/src/master/packages/core/theme
 * https://evergreen.surge.sh/components/colors
 */

import {
  BLACK,
  BLUE,
  BLUE_1,
  BLUE_2,
  GREEN,
  GREENS,
  GREEN_1,
  GREEN_2,
  GREEN_3,
  MAGENTA,
  MUSTARD_1,
  NEUTRAL,
  NEUTRALS,
  ORANGE,
  ORANGE_1,
  OVERLAY_BG,
  PEACH_1,
  PURPLE,
  PURPLES,
  RED,
  REDS,
  RED_1,
  TEAL,
  VIOLET,
  WHITE,
  YELLOW,
  YELLOW_1,
} from './src/Colors';

export {
  BLACK,
  BLUE,
  BLUE_1,
  BLUE_2,
  GREEN,
  GREENS,
  GREEN_1,
  GREEN_2,
  GREEN_3,
  MAGENTA,
  MUSTARD_1,
  NEUTRAL,
  NEUTRALS,
  ORANGE,
  ORANGE_1,
  OVERLAY_BG,
  PEACH_1,
  PURPLE,
  PURPLES,
  RED,
  REDS,
  RED_1,
  TEAL,
  VIOLET,
  WHITE,
  YELLOW,
  YELLOW_1,
};

export default {
  BLUE,
  GREEN,
  MAGENTA,
  NEUTRAL,
  ORANGE,
  PURPLE,
  RED,
  TEAL,
  VIOLET,
  WHITE,
  YELLOW,
};
