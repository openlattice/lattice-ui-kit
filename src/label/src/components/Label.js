import styled from 'styled-components';
import { NEUTRALS } from '../../../colors';

const Label = styled.label`
  color: ${NEUTRALS[0]};
  font-size: 14px;
  font-stretch: normal;
  font-style: normal;
  font-weight: normal;
  letter-spacing: normal;
  margin-top: 10px;
  margin-bottom: 5px;
`;

export default Label;
