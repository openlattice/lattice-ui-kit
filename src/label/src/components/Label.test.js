import React from 'react';

import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';

import Label from './Label';

import { NEUTRALS } from '../../../colors';

describe('Label', () => {
  describe('snapshots', () => {

    test('should match snapshot', () => {
      const wrapper = mount(<Label />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('should render label', () => {
      const wrapper = mount(<Label />);
      expect(wrapper.find('label')).toHaveLength(1);
    });

  });

  describe('styles', () => {
    test('bold prop should apply font-weight: 600', () => {
      const wrapper = mount(<Label bold />);
      expect(wrapper.find(Label)).toHaveStyleRule('font-weight', '600');
    });

    test('subtle prop should apply styles', () => {
      const wrapper = mount(<Label subtle />);
      expect(wrapper.find(Label)).toHaveStyleRule('font-weight', '600');
      expect(wrapper.find(Label)).toHaveStyleRule('color', NEUTRALS[1]);
      expect(wrapper.find(Label)).toHaveStyleRule('font-size', '0.75rem');
      expect(wrapper.find(Label)).toHaveStyleRule('text-transform', 'uppercase');
    });

    test('required prop should apply :after content', () => {
      const wrapper = mount(<Label required>First Name</Label>);
      expect(wrapper.find(Label)).toHaveStyleRule('content', "'*'", {
        modifier: ':after'
      });
    });

    test('stealth prop should set visibility hidden', () => {
      const wrapper = mount(<Label stealth>First Name</Label>);
      expect(wrapper.find(Label)).toHaveStyleRule('visibility', 'hidden');
    });
  });
});
