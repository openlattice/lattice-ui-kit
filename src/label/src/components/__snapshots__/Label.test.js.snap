// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label snapshots should match snapshot 1`] = `
.c0 {
  color: #555e6f;
  display: inline-block;
  font-size: 14px;
  font-stretch: normal;
  font-style: normal;
  font-weight: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin: 5px 5px 5px 0;
}

<Label>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "Label-sc-130fyca-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "color:",
            "#555e6f",
            ";display:inline-block;font-size:14px;font-stretch:normal;font-style:normal;font-weight:",
            [Function],
            ";letter-spacing:normal;margin:5px 5px 5px 0;",
            [Function],
          ],
        },
        "displayName": "Label",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "Label-sc-130fyca-0",
        "target": "label",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <label
      className="c0"
    />
  </StyledComponent>
</Label>
`;
